test/integration_test/basic_dmthin_test.go

```diff
@@ -22,124 +22,259 @@ import (
 	require.True(t, ok)
 	cluster.Name = "test-stc"
 	cluster.Namespace = "kube-system"
 	cluster.Spec.Monitoring.Prometheus.AlertManager.Enabled = false
-	cluster.Annotations = map[string]string{
-		"portworx.io/misc-args": "-T PX-StoreV2",
+	if addDmthinOption {
+		cluster.Annotations = map[string]string{
+			"portworx.io/misc-args": "-T PX-StoreV2",
+		}
 	}
 	return cluster
 }
 
-var testDmthinCases = []types.TestCase{
-	{
-		/* Install DMthin by specifying only metadata device */
-		TestName:        "BasicInstallDmthinWithMetadataDevice",
-		TestrailCaseIDs: []string{""},
-		TestSpec: func(t *testing.T) interface{} {
-			provider := cloud_provider.GetCloudProvider()
-			cluster := defaultDmthinSpec(t)
-			cloudSpec := &corev1.CloudStorageSpec{}
-			cloudSpec.DeviceSpecs = provider.GetDefaultDataDrives()
-			cloudSpec.SystemMdDeviceSpec = provider.GetDefaultMetadataDrive()
-			cluster.Spec.CloudStorage = cloudSpec
-			return cluster
-		},
-		TestFunc: BasicInstallDmthin,
-	},
-	{
-		/* Install DMthin by specifying metadata and kvdb devices */
-		TestName:        "BasicInstallDmthinWithMetadataAndKVDBDevice",
-		TestrailCaseIDs: []string{""},
-		TestSpec: func(t *testing.T) interface{} {
-			provider := cloud_provider.GetCloudProvider()
-			cluster := defaultDmthinSpec(t)
-			cloudSpec := &corev1.CloudStorageSpec{}
-			cloudSpec.DeviceSpecs = provider.GetDefaultDataDrives()
+func getTestName(opt OptionsArr) string {
+	testName := "BasicInstallDmthinWith"
+	if opt[PxStoreV2Present] {
+		testName += "PxStoreV2"
+	}
+	if opt[MetadataPresent] {
+		testName += "Metadata"
+	}
+	if opt[KvdbDevicePresent] {
+		testName += "KVDB"
+	}
+	if opt[JournalPresent] {
+		testName += "Journal"
+	} else if opt[AutoJournalPresent] {
+		testName += "AutoJournal"
+	}
+	return testName
+}
+
+func generateSingleTestCase(o OptionsArr) types.TestCase {
+	testCase := types.TestCase{}
+	testCase.TestName = getTestName(o)
+
+	testCase.TestSpec = func(t *testing.T) interface{} {
+		provider := cloud_provider.GetCloudProvider()
+		cluster := defaultDmthinSpec(t, o[PxStoreV2Present])
+		cloudSpec := &corev1.CloudStorageSpec{}
+		cloudSpec.DeviceSpecs = provider.GetDefaultDataDrives()
+		if o[MetadataPresent] {
 			cloudSpec.SystemMdDeviceSpec = provider.GetDefaultMetadataDrive()
+		}
+		if o[JournalPresent] {
+			cloudSpec.JournalDeviceSpec = provider.GetDefaultJournalDrive()
+		}
+		if o[AutoJournalPresent] {
+			autoStr := "auto"
+			cloudSpec.JournalDeviceSpec = &autoStr
+		}
+		if o[KvdbDevicePresent] {
 			cloudSpec.KvdbDeviceSpec = provider.GetDefaultKvdbDrive()
-			cluster.Spec.CloudStorage = cloudSpec
-			return cluster
-		},
-		TestFunc: BasicInstallDmthin,
-	},
+		}
+
+		cluster.Spec.CloudStorage = cloudSpec
+		return cluster
+	}
+	testCase.TestFunc = BasicInstallDmthin
+	return testCase
 }
 
-func TestDMthinCloudDrivesBasic(t *testing.T) {
-	for _, testCase := range testDmthinCases {
+func generateTestCases(opt OptionsArr, idx uint) []types.TestCase {
```

Comment:
I don't entirely grok what the method is here, but it looks like we're running the same tests twice in a few cases. It would be good to cut these out to save on the 40-ish minutes of tests (given the entire test run took ~3 hours)

```
TestStorageClusterDmthinWithoutPxStoreV2Option/BasicInstallDmthinWithJournal#01
TestStorageClusterDmthinWithoutPxStoreV2Option/BasicInstallDmthinWithKVDBJournal#01
TestStorageClusterDmthinWithoutPxStoreV2Option/BasicInstallDmthinWithMetadataJournal#01
TestStorageClusterDmthinWithoutPxStoreV2Option/BasicInstallDmthinWithMetadataKVDBJournal#01
```

Reply:
The method was to do permutation and I think it generated some dupes. Will fix that.

Reply:
=== RUN   TestStorageClusterDmthinWithoutPxStoreV2Option
time="2023-08-15T22:59:03Z" level=info msg="MYD: BasicInstallDmthinWithJournal"
time="2023-08-15T22:59:03Z" level=info msg="MYD: BasicInstallDmthinWithKVDB"
time="2023-08-15T22:59:03Z" level=info msg="MYD: BasicInstallDmthinWithKVDBJournal"
time="2023-08-15T22:59:03Z" level=info msg="MYD: BasicInstallDmthinWithAutoJournal"
time="2023-08-15T22:59:03Z" level=info msg="MYD: BasicInstallDmthinWithMetadataKVDBAutoJournal"
time="2023-08-15T22:59:03Z" level=info msg="MYD: BasicInstallDmthinWithMetadata"
time="2023-08-15T22:59:03Z" level=info msg="MYD: BasicInstallDmthinWithNothing"
time="2023-08-15T22:59:03Z" level=info msg="MYD: BasicInstallDmthinWithMetadataJournal"
time="2023-08-15T22:59:03Z" level=info msg="MYD: BasicInstallDmthinWithMetadataKVDB"
time="2023-08-15T22:59:03Z" level=info msg="MYD: BasicInstallDmthinWithMetadataKVDBJournal"
time="2023-08-15T22:59:03Z" level=info msg="MYD: BasicInstallDmthinWithMetadataAutoJournal"
time="2023-08-15T22:59:03Z" level=info msg="MYD: BasicInstallDmthinWithKVDBAutoJournal"
--- PASS: TestStorageClusterDmthinWithoutPxStoreV2Option (0.00s)
PASS